src/discussions/learners/LearnersView.test.jsx

```diff
@@ -53,40 +63,197 @@ describe('LearnersView', () => {
         roles: [],
       },
     });
-
+    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
     store = initializeStore();
     Factory.resetAll();
+  });
+
+  async function setUpLearnerMockResponse(
+    count = 3,
+    pageSize = 6,
+    page = 1,
+    username = ['learner-1', 'learner-2', 'learner-3'],
+    searchText,
+  ) {
+    Factory.resetAll();
     const learnersData = Factory.build('learnersResult', {}, {
-      count: learnerCount,
-      pageSize: 6,
+      count,
+      pageSize,
+      page,
     });
-    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
-    axiosMock.onGet(`${coursesApiUrl}${courseId}/activity_stats/`)
+    axiosMock.onGet(learnersApiUrl(courseId))
       .reply(() => [200, learnersData]);
-    const learnersProfile = Factory.build('learnersProfile', {}, {
-      username: ['learner-1', 'learner-2', 'learner-3'],
+
+    axiosMock.onGet(`${getUserProfileApiUrl()}?username=${username.join()}`)
+      .reply(() => [200, Factory.build('learnersProfile', {}, {
+        username,
+      }).profiles]);
+    await executeThunk(fetchLearners(courseId, { usernameSearch: searchText }), store.dispatch, store.getState);
+  }
+
+  async function assignPrivilages() {
+    axiosMock.onGet(`${courseConfigApiUrl}${courseId}/`).reply(200, {
+      learners_tab_enabled: true,
+      user_is_privileged: true,
+    });
+    axiosMock.onGet(`${courseConfigApiUrl}${courseId}/settings`).reply(200, {});
+    await executeThunk(fetchCourseConfig(courseId), store.dispatch, store.getState);
+  }
+
+  it('Learners tab is disabled by default', async () => {
+    await setUpLearnerMockResponse();
+    await act(async () => {
+      await renderComponent();
     });
-    axiosMock.onGet(`${userProfileApiUrl}?username=learner-1,learner-2,learner-3`)
-      .reply(() => [200, learnersProfile.profiles]);
-    await executeThunk(fetchLearners(courseId), store.dispatch, store.getState);
+    expect(screen.queryByText(/Last active/i)).toBeFalsy();
   });
 
-  describe('Basic', () => {
-    test('Learners tab is disabled by default', async () => {
+  it('Learners tab is enabled', async () => {
+    await setUpLearnerMockResponse();
+    await assignPrivilages();
+    await act(async () => {
+      await renderComponent();
+    });
+  });
+
+  it('Most activity should be selected by default for the non-moderator role.', async () => {
+    await setUpLearnerMockResponse();
```

Comment:
setUpLearnerMockResponse() function is used in every function without any param modification. so instead of making a function just mock API In beforeEach.